identity-wallet/src/state/persistence.rs

```diff
@@ -22,6 +23,10 @@ pub async fn load_state() -> anyhow::Result<AppState> {
 pub async fn save_state(app_state: &AppState) -> anyhow::Result<()> {
     let state_file = STATE_FILE.lock().unwrap().clone();
     let mut file = File::create(state_file).await?;
+
+    let mut json_app_state = serde_json::to_value(app_state)?;
+    json_app_state["credentials"] = serde_json::Value::Array(Vec::new());
+
```

Comment:
*is not persisted

Reply:
Maybe you can clone app_state, then remove credentials on rust side before storing. That way you let Rust and serde do the work.

Reply:
I guess perhaps each Feature should have the ability to set its persistence strategy, either pure json file or stronghold. So `pub async fn save_state(app_state: &AppState)` would be `pub async fn save_state(app_state: Box<dyn FeatState>)` or something similar.

Perhaps even better it could/should be implemented via the `FeatTrait`. (for another time though, I think the current solution, although not super clean, simply does the job).